docs/dev-guide/policies/style.rst

```diff
@@ -11,11 +11,33 @@ length limit is 100 characters, which we chose to prevent line wrapping on GitHu
 In-code Documentation
 ---------------------
 
-Document your functions using the markup described
-`here <http://sphinx-doc.org/markup/desc.html#info-field-lists>`_.
-If it's worth having a function, it's worth taking 1 minute to describe what it
-does, define each parameter, and define its return value. This saves a
-tremendous amount of time for the next person who looks at your code.
+All new code must have a doc block that describes what the function or class does,
+describes each of the parameters, lists the parameter types (fully qualified please), lists
+exceptions that can be raised, describes the return value, and lists the type of the return value.
+
+Example::
+
+  def update(title, author, book):
+      """
+      Updates information about a book and returns the updated object.
+
+      :param title: new title the book should be assigned
+      :type  title: basestring
+      :param author: new author that the book should be assigned
+      :type  author: fully.qualified.author
+      :param book: book object that will be updated
+      :type  book: fully.qualified.book
+
+      :raises BookNotFound: if book is not found
+
+      :return: updated book object
+      :rtype:  fully.qualified.book
```

Comment:
if you decided to describe the exception and return value, please, for the consistency mention them also in the example's code